### PR TITLE
 Fix #7771: Support [TYPE] in signature

### LINKED ISSUE #7771: Support [TYPE] in signature
From support mail:
To add cache type for the log signatures so that it would say something like: Thanks to [OWNER]  for this [TYPE].

### PR DESCRIPTION
**Pull request by bmuzzin**
Add [TYPE] as a signature template, and also as a logging insertion option.

### CODE REVIEW

**Comment by bmuzzin** [comment]
Will reopen with style fixes.

**Comment by SammysHP** [comment]
Hi @bmuzzin,

thank you for the pull request! To make things easier to understand (when reading the commit history), please reword your commit and PR message to something like `Fix #7771: Support [TYPE] in signature`. This will also close the issue automatically once the PR got merged.

**Comment by Lineflyer** [comment]
ok to test

**Comment by bmuzzin** [comment]
Okay, I've updated the commit comment as you suggested. I'll and start working from a different branch from now on... while I'm not new to git, I'm new to the whole github side of things, so excuse my ignorance on these sorts of things. Thanks!

**Comment by Lineflyer** [comment]
retest this please

File: main/src/cgeo/geocaching/log/LogTemplateProvider.java [hunk 1]
```diff
@@ -285,6 +286,17 @@ public String getValue(final LogContext context) {
                 return StringUtils.EMPTY;
             }
         });
+        templates.add(new LogTemplate("TYPE",  R.string.init_signature_template_type) {
+            @Override
+            public String getValue(final LogContext context) {
+                final Geocache cache = context.getCache();
+                if (cache != null) {
+                    final CacheType cacheType = cache.getType();
+                    return cacheType.pattern;
```
Comment (moving-bits) [inline main/src/cgeo/geocaching/log/LogTemplateProvider.java:295]: `CacheType.pattern` is the non-translatable string used by GPX files etc. Should use `CacheType.stringId` instead AFAICS, which is the translated type identifier.

Comment (bmuzzin) [reply main/src/cgeo/geocaching/log/LogTemplateProvider.java]: Looks like the translated stringId is private, but comes through translated with getL10n() - not sure what the function is named for? .. But seems to be used in many places to get the translated name.

**Comment by fm-sys** [comment]
retest this please

File: main/src/cgeo/geocaching/log/LogTemplateProvider.java (hunk 1 above)
Comment (moving-bits) [reply main/src/cgeo/geocaching/log/LogTemplateProvider.java]: @bmuzzin
Thanks for updating the PR.
BTW: "L10n" stands for "Localization" which starts with a "L", ends with a "n" and has 10 letters in between...

**Pull request merged**